File: Candy/js_candy/candy19.js
// 編號：CANDY-019
// 程式語言：JavaScript
// 題目：檢查是否為某個數字的平方數

function isSquare(num) {
    // 實作寫在這裡
    return Number.isInteger(Math.sqrt(num));
  }
  
  console.log(isSquare(0)) // true
  console.log(isSquare(4)) // true
  console.log(isSquare(5)) // false
  console.log(isSquare(100)) // true
  console.log(isSquare(-4)) // false
  console.log(isSquare(-1)) // false

File: Candy/rb_candy/candy19.rb
# 編號：CANDY-019
# 程式語言：Ruby
# 題目：檢查是否為某個數字的平方數

def is_square num
  # 實作寫在這裡
  return false if num < 0
  
  Math.sqrt(num) % 1 == 0
end

p is_square(0) # true
puts is_square(4) # true
puts is_square(5) # false
puts is_square(100) # true
puts is_square(-4) # false
puts is_square(-1) # false